packages/constants/src/chains.ts

```diff
@@ -1,17 +1,32 @@
-// Canonical data source: https://github.com/cosmos/chain-registry/tree/master
+import { AssetId } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb';
+
 export interface Chain {
   displayName: string;
   chainId: string;
   ibcChannel: string;
   iconUrl: string;
+  nativeAssets: AssetId[];
```

Comment:
ibc token ids will be added here when we figure out what those assets are TBD

Reply:
We should not have an explicit list of native assets per chain, we already have an asset registry and we can figure out the chain from the channel in the denom

Reply:
Can you provide a few examples of what these denoms look like?

Reply:
Ah, wait, is it just a normal ibc asset like:

```
ibc/14F9BC3E44B8A9C1BE1FB08980FAB87034C9905EF17CF2F5008FC085218811CC
```

If so, that's a hash of: `transfer/channel-141/uosmo`. So that means we'd need to loop over our asset registry, doing the hashing of each one until we find a match in order to see if that ibc asset belongs to the particular channel the user selected. Does that sound right to you?

Reply:
Nope — what I was trying to flag is that unlike ibc-go, we don't do the ibc/HASH thing. So the denom of that IBC asset is just

transfer/channel-141/uosmo

and its asset ID is the asset ID for that denom.

The way to think about this is: in the cosmos sdk, they started with coin denoms, then later realized that there should be asset IDs, and retrofitted an asset ID mechanism _inside_ the denom string. But in Penumbra, we had to have asset IDs from the start to record values in the shielded pool, so we made the asset ID be a hash of the denom from the start. And then we had to build a system for getting metadata based on asset ID as the base case.

The upshot is that handling IBC assets _inside_ Penumbra is way simpler, it's just like metadata of any other asset, IBC or not, and we can inspect the path directly in the denom.